src/ucp/core/ucp_ep.c

```diff
@@ -56,10 +57,17 @@ static ucs_stats_class_t ucp_ep_stats_class = {
 
 void ucp_ep_config_key_reset(ucp_ep_config_key_t *key)
 {
+    ucp_lane_index_t i;
     memset(key, 0, sizeof(*key));
     key->num_lanes        = 0;
+    for (i = 0; i < UCP_MAX_LANES; ++i) {
+        key->lanes[i].rsc_index    = UCP_NULL_RESOURCE;
+        key->lanes[i].proxy_lane   = UCP_NULL_LANE;
+        key->lanes[i].dst_md_index = UCP_MAX_MDS;
```

Comment:
why not zero or UCP_NULL_RESOURCE?

Reply:
`0` is a valid value for rsc/lane/md index, that's buggy. `UCP_NULL_RESOURCE`, `UCP_NULL_LANE` and `UCP_MAX_MDS` technically have the same value now but can be changed later, so better to initialize with correct constant every variable.